temporal-sdk/src/main/java/io/temporal/worker/tuning/ResourceController.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2022 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ * Copyright (C) 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this material except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.temporal.worker.tuning;
+
+import io.temporal.common.Experimental;
+import java.time.Instant;
+import java.util.concurrent.locks.ReentrantLock;
+
+/**
+ * Is used by {@link ResourceBasedSlotSupplier} and {@link ResourceBasedTuner} to make decisions
+ * about whether slots should be handed out based on system resource usage.
+ *
+ * @param <RI> How this controller obtains system resource information
+ */
+@Experimental
+public class ResourceController<RI extends SystemResourceInfo> {
```

Comment:
I don't think there's any value in a generic here. You only care about the interface, so just reference `SystemResourceInfo` everywhere.

Reply:
Yeah, good point